packages/nuxi/src/utils/prepare.ts

```diff
@@ -32,6 +32,7 @@ export const writeTypes = async (nuxt: Nuxt) => {
       ...nuxt.options.typescript.includeWorkspace && nuxt.options.workspaceDir !== nuxt.options.rootDir ? [join(relative(nuxt.options.buildDir, nuxt.options.workspaceDir), '**/*')] : []
     ],
     exclude: [
+      relative(nuxt.options.buildDir, nuxt.options.serverDir),
```

Comment:
To avoid making it a breaking change for type support shall we try to detect `server/tsconfig.json` and optionally ignore (also make a warning to enable file?)